modules/lang-painless/src/main/java/org/elasticsearch/painless/lookup/PainlessLookupBuilder.java

```diff
@@ -937,6 +1089,11 @@ public void addPainlessBinding(Class<?> targetClass, String methodName, Class<?>
         }
 
         String painlessMethodKey = buildPainlessMethodKey(methodName, constructorTypeParametersSize + methodTypeParametersSize);
+
+        if (painlessMethodKeysToPainlessStatics.containsKey(painlessMethodKey)) {
+            throw new IllegalArgumentException("binding and static cannot have the same name [" + methodName + "]");
```

Comment:
static -> static methods?

Reply:
Renamed to imported methods for all of these related to this feature.  Will rename bindings to classbindings in the instancebindings PR or a separate PR.